app/models/insights_hit.rb

```diff
@@ -1,6 +1,15 @@
+module ScopedSearch::QueryLanguage::Tokenizer
+  def tokenize_keyword(&block)
+    keyword = current_char
+    # remove "|" from this regex so it works with Insights rule IDs
+    keyword << next_char while /[^=~<>\s\&\)\(,]/ =~ peek_char
+    KEYWORDS.has_key?(keyword.downcase) ? yield(KEYWORDS[keyword.downcase]) : yield(keyword)
+  end
+end
+
 class InsightsHit < ApplicationRecord
   include ::Authorizable
-  belongs_to :host
+  belongs_to :host, class_name: 'Host::Base'
```

Comment:
Side note: `InsightsHit.search_for(xxx).to_sql` wouldn't work before I added this. This alone took quite a bit of debugging.